### DISCUSSION TITLE
Biggest MkDocsMaterial-powered repositories

### DISCUSSION THREAD

**Discussion by couleurm** [squidfunk/mkdocs-material#6557] [General]
https://github.com/squidfunk/mkdocs-material/network/dependents?dependent_type=REPOSITORY

This link allows us to see all repositories which use mkdocs-material (e.g in their `requirements.txt`)

Wouldn't it be cool to have a list of the most starred repositories using it?

**Comment by squidfunk** [comment]
Jup, correct. The link is also listed on our [GitHub homepage](https://github.com/squidfunk/mkdocs-material):

<img width="323" alt="Bildschirm­foto 2023-12-28 um 16 59 00" src="https://github.com/squidfunk/mkdocs-material/assets/932156/dc51ab9d-a893-49fe-98b5-9f3f56e5097a">

**Reply by alexvoss** [reply]
Looks like the detailed info is not available through the API but [someone has written a scraper...](https://pypi.org/project/github-dependents-info/) **Edit**: Having some problems getting it to work, though.

**Reply by alexvoss** [reply]
Right, coming back to this. I think there is something wrong with the long form argument handling. This now worked for me:

```
github-dependents-info -r squidfunk/mkdocs-material -j > data.json
```

with some post-processing:

```
Significant-Gravitas/AutoGPT (155540)
pola-rs/polars (22551)
juanfont/headscale (17189)
kubernetes/ingress-nginx (16138)
beekeeper-studio/beekeeper-studio (14190)
ydataai/ydata-profiling (11611)
cpacker/MemGPT (6994)
cilium/ebpf (5306)
apache/iceberg (5060)
online-ml/river (4573)
neutrinojs/neutrino (3948)
outlines-dev/outlines (3837)
shroominic/codeinterpreter-api (3474)
weaveworks/ignite (3472)
blacklanternsecurity/bbot (3042)
Avaiga/taipy (2738)
guardrails-ai/guardrails (2631)
hlorus/CAD_Sketcher (2301)
saket/cascade (1916)
sudoskys/StableDiffusionBook (1710)
```

That would be the ones for the public version, of course. Also, the ones that are caught by the heuristics applied by GitHub.

**Reply by squidfunk** [reply]
At least [FastAPI](https://github.com/tiangolo/fastapi) (66k stars) is missing here. [They are using the community edition in non-CI environments](https://github.com/tiangolo/fastapi/blob/040ad986d48bb9a400de804f7f25abb856d85e1a/requirements-docs.txt#L3).

**Comment by nvuillam** [comment]
@couleurm @squidfunk the PR is ready :)

https://github.com/squidfunk/mkdocs-material/pull/6574

![image](https://github.com/squidfunk/mkdocs-material/assets/17500430/7ad26007-7a16-4a39-a89b-185c7fbe7db6)

**Reply by couleurm** [reply]
super cool! is this up anywhere? github cant load the preview for this 6mbn file :(

**Reply by squidfunk** [reply]
If it's up somewhere, we're happy to link it from our docs ✌️

**Reply by nvuillam** [reply]
It is up here -> https://github.com/nvuillam/mkdocs-material-github-dependents-info

But GitHub does not want to display a 6mb markdown file... we'll have to... add a mkdocs.yml to publish it as GitHub page ;)

**Reply by squidfunk** [reply]
Could you maybe patch the script to introduce pagination, say, 100 projects per page? It would be more mobile friendly (think slow connections) and you could also preview it on GitHub. I'm also still thinking of a good spot to link to it.

**Reply by nvuillam** [reply]
@squidfunk I will do that, and even generate the "Used by" image so it could be referred to statically from your README.md :)
